### DISCUSSION TITLE
Unsubscribe multiple nodes at once

### DISCUSSION THREAD

**Discussion by Ajinkya-Suranis** [FreeOpcUa/opcua-asyncio#1560] [Q&A]
I'm working on a program in which datachange subscriptions can be added/deleted on multiple nodes, as per request from user. However, when unsubscribing, the subscription handle is passed, which means that all the nodes which were subscribed, will be unsubscribed. There seems to be no way to selectively unsubscribe few nodes.
As a workaround, I'm creating a separate handle for datachange subscription for each node, so that if some nodes need to be unsubscribed, then I'll iteratively unsubscribe them using corresponding handles.
Is there any better way to achive this?

**Comment by schroeder-** [comment]
Have a look at this method https://github.com/FreeOpcUa/opcua-asyncio/blob/3268568df3b2c314920d97bb059cc48b3a98d5a1/asyncua/common/subscription.py#L484C1-L488C87
This method removes only the monitored item nodes you want to remove from the subscription,
